src/Microsoft.TestPlatform.AdapterUtilities/ManagedNameUtilities/ManagedNameHelper.Reflection.cs

```diff
@@ -265,7 +266,17 @@ bool Filter(MemberInfo mbr, object? param)
         methods = type.GetRuntimeMethods().Where(m => Filter(m, null)).ToArray();
 #endif
 
-        return (MethodInfo?)methods.SingleOrDefault();
+        MethodInfo? method = null;
+        if (methods.Length == 1)
+        {
+            method = (MethodInfo?)methods[0];
+        }
+        else if (methods.Length > 1)
+        {
+            method = (MethodInfo?)methods.SingleOrDefault(i => i.DeclaringType == type);
+        }
+
+        return method;
```

Comment:
Nit: Switch pattern would be easier to read. I am suggesting something without having tested it so there might be some small fixes to do:

```suggestion
        return methods.Length switch        
        {
            1 => (MethodInfo?)methods[0],
            > 1 =>(MethodInfo?)methods.SingleOrDefault(i => i.DeclaringType == type),
            _ => null,
        };
```